src/katsdpcalproc/pointing.py

```diff
@@ -16,72 +15,71 @@
+        containing gains
+    offsets : :class: 'list'
+        list of requested (x, y) pointing offsets co-ordinates relative to target,
+        in degrees.
+    ants: list of :class: 'katpoint.Antenna'
+        A list of antenna objects for fitted beams
+    track_duration : :class: 'float',
+        Duration of each pointing track, in seconds
+    centre_freq, bandwidth, n_channels: :class: 'float'
+        centre frequency, bandwidth and number of channels
+    pols: :class: 'list'
+        A list containing polarisations, eg, ["h","v"]
+    num_chunks: :class: 'float'
+        Group the frequency channels into this many sections to
         obtain pointing fits (default =16)
-    ants: A list containing <class: katpoint.Antenna> objects for each
-        antenna used in the observation
-    centre_freq, bandwidth, no_channels: floats, centre frequency, bandwidth
-        and number of frequency channels
-    pols: list, A list containing polarisations, eg, ["h","v"]
 
     Returns
     -------
-    data_points : dict mapping receptor index to (x, y, freq, gain, weight) seq
-            Complex gains per receptor, as multiple records per offset and
-            frequency chunk ie. len(data_points)=63, len(data_points[i])=
-            Num_chunks*no.offsets, len(data_points[i][j]=5)
+    data_points : :class: 'dict'
+        dict mapping receptor index to (x, y, freq, gain, weight) seq
+        Complex gains per receptor, as multiple records per offset and
+        frequency chunk ie. len(data_points)=63, len(data_points[i])=
+        num_chunks*n_offsets, len(data_points[i][j]=5)
     """
-
-    if no_channels % NUM_CHUNKS != 0:
-        raise NotMUltipleError("NUM_CHUNKS is not a multiple of number of channels")
+    if bp_gains.shape != (len(offsets), n_channels, len(pols), len(ants)):
+        raise IncorrectShape("""bp_gains must have shape (n_offsets, n_channels,
+                             n_polarizations, n_antennas)""")
     # Calculating chunk frequencies
-
-    channel_freqs = centre_freq + ((np.arange(no_channels) - no_channels / 2)
-                                   * (bandwidth / no_channels))
-
-    chunk_freqs = channel_freqs.reshape(NUM_CHUNKS, -1).mean(axis=1)
+    channel_freqs = centre_freq + ((np.arange(bp_gains.shape[1]) - bp_gains.shape[1] / 2)
+                                   * (bandwidth / bp_gains.shape[1]))
+    chunk_freqs = channel_freqs.reshape(num_chunks, -1).mean(axis=1)
     data_points = {}
-
-    for i, e, f in zip(offsets, bp_gains, gains):
-
+    for offset, offset_bp_gain, offset_gain in zip(offsets, bp_gains, gains):
         for a, ant in enumerate(ants):
-
-            pol_gain = np.zeros(NUM_CHUNKS)
-            pol_weight = np.zeros(NUM_CHUNKS)
+            pol_gain = np.zeros(num_chunks)
+            pol_weight = np.zeros(num_chunks)
             # Iterate over polarisations (effectively over inputs)
-
             for pol in range(0, len(pols)):
```

Comment:
There are more of these superfluous 0s in the code. Replace `range(0, len(x))` with `range(len(x))`.